frontend/server/src/main/java/org/pytorch/serve/job/GRPCJob.java

```diff
@@ -96,30 +117,62 @@ public void response(
             Map<String, String> responseHeaders) {
         ByteString output = ByteString.copyFrom(body);
         WorkerCommands cmd = this.getCmd();
+        Gson gson = new Gson();
+        String jsonResponse = output.toStringUtf8();
+        JsonObject jsonObject = gson.fromJson(jsonResponse, JsonObject.class);
 
         switch (cmd) {
             case PREDICT:
             case STREAMPREDICT:
             case STREAMPREDICT2:
-                ServerCallStreamObserver<PredictionResponse> responseObserver =
-                        (ServerCallStreamObserver<PredictionResponse>) predictionResponseObserver;
-                cancelHandler(responseObserver);
-                PredictionResponse reply =
-                        PredictionResponse.newBuilder().setPrediction(output).build();
-                responseObserver.onNext(reply);
-                if (cmd == WorkerCommands.PREDICT
-                        || (cmd == WorkerCommands.STREAMPREDICT
-                                && responseHeaders
-                                        .get(RequestInput.TS_STREAM_NEXT)
-                                        .equals("false"))) {
-                    responseObserver.onCompleted();
-                    logQueueTime();
-                } else if (cmd == WorkerCommands.STREAMPREDICT2
-                        && (responseHeaders.get(RequestInput.TS_STREAM_NEXT) == null
-                                || responseHeaders
-                                        .get(RequestInput.TS_STREAM_NEXT)
-                                        .equals("false"))) {
-                    logQueueTime();
+                // condition for OIP grpc ModelInfer Call
+                if (ConfigManager.getInstance().isOpenInferenceProtocol() && isResponseStructureOIP(jsonObject)) {
```

Comment:
This data structure check can be broken if some model output meet this requirement but the request is not kserve request. And also it is not efficient to check each response. I prefer creating a set of. workercommand for kserve to make the code efficient and be easily maintained.